### PR TITLE
`cosign` v1.13.1 and fix the `creating root cert pool` / `one of verifier or root certs is required` errors

### PR DESCRIPTION
**Pull request by mathieu-benoit**
Signed-off-by: Mathieu Benoit <mathieu-benoit@hotmail.fr>

Bump `cosign` from v1.3.1 to v1.13.1.

Fixing the errors:
```
verify signature for: devopps/alpine:signed
2023/02/07 00:34:19 http: panic serving 10.0.3.172:34976: creating root cert pool: retrieving trusted root; local cache may be corrupt: initializing root client: tuf: failed to decode root.json: encoding/hex: invalid byte: U+002D '-'
goroutine 9 [running]:
net/http.(*conn).serve.func1()
	/usr/local/go/src/net/http/server.go:1801 +0xb9
panic({0x1b9fa40, 0xc000257c20})
	/usr/local/go/src/runtime/panic.go:1047 +0x266
github.com/sigstore/cosign/cmd/cosign/cli/fulcio/fulcioroots.initRoots()
	/go/pkg/mod/github.com/sigstore/cosign@v1.3.1/cmd/cosign/cli/fulcio/fulcioroots/fulcioroots.go:67 +0x235
github.com/sigstore/cosign/cmd/cosign/cli/fulcio/fulcioroots.Get.func1()
	/go/pkg/mod/github.com/sigstore/cosign@v1.3.1/cmd/cosign/cli/fulcio/fulcioroots/fulcioroots.go:45 +0x17
sync.(*Once).doSlow(0xc0003ac138, 0x18)
	/usr/local/go/src/sync/once.go:68 +0xd2
sync.(*Once).Do(...)
	/usr/local/go/src/sync/once.go:59
github.com/sigstore/cosign/cmd/cosign/cli/fulcio/fulcioroots.Get()
	/go/pkg/mod/github.com/sigstore/cosign@v1.3.1/cmd/cosign/cli/fulcio/fulcioroots/fulcioroots.go:44 +0x31
github.com/sigstore/cosign/cmd/cosign/cli/fulcio.GetRoots(...)
	/go/pkg/mod/github.com/sigstore/cosign@v1.3.1/cmd/cosign/cli/fulcio/fulcio.go:197
main.validate({0x218ee50, 0xc000562000}, 0xc000560300)
	/go/src/github.com/developer-guy/cosign-gatekeeper-provider/provider.go:72 +0x408
net/http.HandlerFunc.ServeHTTP(0x7fabe0ff4aa0, {0x218ee50, 0xc000562000}, 0xc000562000)
	/usr/local/go/src/net/http/server.go:2046 +0x2f
net/http.(*ServeMux).ServeHTTP(0x0, {0x218ee50, 0xc000562000}, 0xc000560300)
	/usr/local/go/src/net/http/server.go:2424 +0x149
net/http.serverHandler.ServeHTTP({0x2182d08}, {0x218ee50, 0xc000562000}, 0xc000560300)
	/usr/local/go/src/net/http/server.go:2878 +0x43b
net/http.(*conn).serve(0xc00041c3c0, {0x2199b40, 0xc000651980})
	/usr/local/go/src/net/http/server.go:1929 +0xb08
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:3033 +0x4e8
verify signature for: devopps/alpine:signed
one of verifier or root certs is required
```
I'm still getting this message with my own public signed image:
```
verify signature for: devopps/alpine:signed
no matching signatures:
no certificate found on signature
```
But that's the same message when I run this command `COSIGN_EXPERIMENTAL=1 cosign verify devopps/alpine:signed`:
```
Error: no matching signatures:
no certificate found on signature
main.go:62: error during command execution: no matching signatures:
no certificate found on signature
```
But at least we are making progress with this PR.

### CODE REVIEW

**Comment by mathieu-benoit** [comment]
I'm able to build locally this container image and test it in my Kubernetes cluster, no problem.

One of the `Build / test` check in the CI is green/successful.

Where I NEED HELP is to help me understand and fix the second `Build / test` check and the `Test / test` check in CI... not sure what I should do. Anyone here to help? Thanks!

File: provider.go [hunk 1]
```diff
@@ -81,6 +83,24 @@ func validate(w http.ResponseWriter, req *http.Request) {
...
+	if err != nil {
+		sendResponse(nil, fmt.Sprintf("getting root certs: %v", err), w)
+		return
+	}
+
+	fulcioIntermediates, err := fulcio.GetIntermediates()
+	if err != nil {
+		sendResponse(nil, fmt.Sprintf("getting Fulcio intermediates: %v", err), w)
+		return
+	}
```
Comment (Dentrax) [inline provider.go:102]: Can you please move this initialization logic to `main()` function? By doing so, we can avoid to init a new rekor client on every request. So it should be singleton instance. Maybe you can move the required variables to global var: `RekorClient`, `RootCerts`, `IntermediateCerts`.

**Review by Dentrax — CHANGES_REQUESTED** [review]
Thanks for the PR @mathieu-benoit!

The error is because of the version of Go; underlaying `go-tuf` package uses `strings.Cut`, which is introduced with `go1.18`. I think we should prioritize merging #24 first to make it pass.

File: provider.go (hunk 1 above)
Comment (mathieu-benoit) [reply provider.go:102]: Good point, yeah saw that in your other PR. I will do it, thanks!

**Comment by mathieu-benoit** [comment]
Thanks @Dentrax.

> I think we should prioritize merging https://github.com/sigstore/cosign-gatekeeper-provider/pull/24 first to make it pass.

I agree with you.

**Comment by mathieu-benoit** [comment]
Woot! Woot! `All checks have passed` with #24.

I will soon work on the changes requested.

**Comment by Dentrax** [comment]
Good news! Looking for your changes. I will be sneaking around the merge button.

**Comment by mathieu-benoit** [comment]
We should be could to go, thanks for your review @Dentrax!

File: provider.go (hunk 1 above)
Comment (mathieu-benoit) [reply provider.go]: Done, let me know if there is anything else, thanks!

**Review by Dentrax — APPROVED** [review]
Thanks!

**Pull request merged**